packages/block-library/src/post-comments-form/edit.js

```diff
@@ -39,7 +40,49 @@ export default function PostCommentsFormEdit( {
 		} ),
 	} );
 
-	const isInSiteEditor = postType === undefined || postId === undefined;
+	const isSiteEditor = postType === undefined || postId === undefined;
+
+	const { defaultCommentStatus } = useSelect(
+		( select ) =>
+			select( blockEditorStore ).getSettings()
+				.__experimentalDiscussionSettings
+	);
+
+	const postTypeSupportsComments = useSelect( ( select ) =>
+		postType
+			? !! select( coreStore ).getPostType( postType )?.supports.comments
+			: false
+	);
+
+	let warning = false;
```

Comment:
By default we don't display a warning, like we do in the Post Comments block:

https://github.com/WordPress/gutenberg/blob/54fd3bfd349fbe24485bb07899dd25d3aee7fa10/packages/block-library/src/post-comments/edit.js#L53-L55

As the Post Comments Form **is included** in the Comments Query Loop, it would not make sense to say that you should "replace it with the Comments Query Loop". 😅 

The form rendered in the editor should also be close enough style-wise to the one rendered on the frontend so that we don't need to warn about it.